instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -256,6 +257,13 @@ def _trace_failure(*args, **kwargs):
             return
 
         if ex is not None:
+            # Unwrap the actual exception wrapped by
+            # Celery's `ExceptionInfo` and report that.
+            if (
+                isinstance(ex, ExceptionInfo)
```

Comment:
What version of celery did you test? for celery 5.3 the ExceptionInfo changed and now the ex.exception is ExceptionWithTraceBack, so maybe in this case we need the exception field inside the ExceptionWithTraceBack

Reply:
Right! I'll amend this for newer version. I'll also import these from `billiard.einfo`, since it seems like newer versions of Celery don't expose them from `celery.datastructures`.